app/models/user.rb

```diff
@@ -35,6 +35,10 @@ def last_submission_date
     assignments.last.try(&:updated_at)
   end
 
+  def last_submitted_exercise
+    assignments.last.try(&:exercise)
```

Comment:
User already has a - denormalized - `last_exercise` stored in DB. So I think this new method is not necessary.